pkg/internal/integration_test_support/telemetry.go

```diff
@@ -0,0 +1,92 @@
+package integrationtestsupport
+
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v3"
+
+	prefab "github.com/prefab-cloud/prefab-cloud-go/pkg"
+	"github.com/prefab-cloud/prefab-cloud-go/pkg/internal/contexts"
+	"github.com/prefab-cloud/prefab-cloud-go/pkg/internal/telemetry"
+	prefabProto "github.com/prefab-cloud/prefab-cloud-go/proto"
+)
+
+type TelemetryTestCaseYaml struct {
+	Aggregator   string      `yaml:"aggregator"`
+	Endpoint     string      `yaml:"endpoint"`
+	Data         interface{} `yaml:"data"`
+	ExpectedData interface{} `yaml:"expected_data"`
+	CaseName     string      `yaml:"name"`
+	Client       string      `yaml:"client"`
+	Function     string      `yaml:"function"`
+}
+
+type TelemetryTest struct {
+	Cases []TelemetryTestCaseYaml `yaml:"cases"`
+}
+
+type TelemetryTestCase struct {
+	Contexts   TestCaseContexts
+	Aggregator string
+	TestName   string
+	Err        error
+	Yaml       TelemetryTestCaseYaml
+	RawYaml    []byte
+}
+
+func (t TelemetryTestCase) GetGlobalContexts() *contexts.ContextSet {
+	return t.Contexts.Global
+}
+
+func (t TelemetryTestCase) GetBlockContexts() *contexts.ContextSet {
+	return t.Contexts.Block
+}
+
+func (t TelemetryTestCase) GetClientOverrides() *ClientOverridesYaml {
+	return nil
+}
+
+type TelemetryTestHarness interface {
+	GetOptions() []prefab.Option
+	GetExpectedEvent() (*prefabProto.TelemetryEvent, error)
+	Exercise(*prefab.ContextBoundClient) error
+	MassagePayload(events *prefabProto.TelemetryEvents) *prefabProto.TelemetryEvents
+}
```

Comment:
Each telemetry test works off these four methods. Options for client setup, Massaging the expected event YAML into protos, Exercising the test based on the input, and Massaging the Payload (e.g. to exclude things we intentionally don't care about in the YAML test like configId and SelectedValue)